text/0002-searchable-display-attributes-behaviors.md

```diff
@@ -0,0 +1,255 @@
+
+> When the searchable attributes value is set to `["*"]` the priority of the documents attributes is undefined.
+
+By default searchable attributes should be in the order of the first indexed document, for example:
+
+```json
+{
+  "title": "many the fish",
+  "description": "it is awfull to be liked by many"
+}
+```
+The default `searchableAttributes` order should be `["title", "description"]` where `"title" > "description"`.
+
+TBD: What happens if the order is changed and wildcard is set back to `["*"]`?
+- undefined behaviour: the order when using `["*"]` can't be relied upon
+- some default value: which? Why?
+- Dissociate order and existence.
+
+#### Pre-existing attributes
+
+Another issue with the current functioning of the attributes resides in the way we register the attributes. If a user adds, for example, attributes to the `facetAttributes` before indexing any documents, then this field will be registered as a known field, and when the user adds eventually indexes his documents, the field previously registered is not add to the searchable and displayed attributes.
+
+This means that there should be a distinction between the creation of an `fieldId` (when a field is first registered), and it's usage/position in `searchableAttributes` and `displayedAttributes`.
+What that means is that when we index documents, we must check if a field already has a position, and if not, give it a position (after adding it to the fieldmap) according to its position in the document.
+
+##### caveat
+- If a new document contains a new field at a position for which we already have a field:
+```json
+[
+	{
+		"id": 1,
+		"title": "the title"
+	},
+	{
+
+		"id": 1,
+		"titre": "le titre"
+	}
+]
+```
+
+- To make sure that the `searchableAttributes` and `displayedAttributes` are in sync, the must come from the same source of information; this should be the field_position.
+ TBD: What should be the position of `titre`?
+	- same position as `title` (thus allowing attributes to exists at the same position like Algolia)
+	- added at the last position
+
+#### `searchableAttributes` & `displayedAttributes`
+
+Being searchable and displayed are 2 different concept, and there should not be interactions between the two: a field can be searchable and not displayed, or displayed and not searchable, etc.
+
+#### add a new configuration `attributesPosition`?
+
+Adding a new configuration `attributesPosition` that link all attributes present in a document  to a position makes it possible to keep track of the position without caring about whether a field is searchable or displayed. This also allows to keep track of all the fields that have been found in documents in the database.
+
+
+The default order is then **always** the order a field was in a document when it was first found.
+
+##### Content of `attributesPosition`
```

Comment:
The absence of a field in `searchableAttributes` does not impact what will be added to `attributesPosition`?

Reply:
no it doesn't, all fields that are present in any documents are given an attribute position. This way you can add or remove an attribute position, and this will not impact it's position in the documents. No data is ever lost.